pkg/ucp/frontend/api/server.go

```diff
@@ -190,51 +192,88 @@ func (s *Service) Initialize(ctx context.Context) (*http.Server, error) {
 
 // configureDefaultPlanes reads the configuration file specified by the env var to configure default planes into UCP
 func (s *Service) configureDefaultPlanes(ctx context.Context) error {
+	for _, plane := range s.options.InitialPlanes {
+		err := s.createPlane(ctx, plane)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *Service) createPlane(ctx context.Context, plane rest.Plane) error {
+	body, err := json.Marshal(plane)
+	if err != nil {
+		return err
+	}
+
+	resourceID, err := resources.ParseScope(plane.ID)
+	if err != nil {
+		return err
+	}
+
+	if len(resourceID.ScopeSegments()) != 1 {
+		return fmt.Errorf("invalid plane ID: %s", plane.ID)
+	}
+
 	db, err := s.storageProvider.GetStorageClient(ctx, "ucp")
 	if err != nil {
 		return err
 	}
 
-	for _, plane := range s.options.InitialPlanes {
-		body, err := json.Marshal(plane)
-		if err != nil {
-			return err
-		}
+	opts := armrpc_controller.Options{
+		StorageClient: db,
+	}
 
-		opts := armrpc_controller.Options{
-			StorageClient: db,
-		}
-		planesCtrl, err := defaultoperation.NewDefaultSyncPut(opts,
-			armrpc_controller.ResourceOptions[datamodel.Plane]{
-				RequestConverter:  converter.PlaneDataModelFromVersioned,
-				ResponseConverter: converter.PlaneDataModelToVersioned,
-			},
-		)
-		if err != nil {
-			return err
-		}
+	var ctrl armrpc_controller.Controller
+	switch strings.ToLower(resourceID.ScopeSegments()[0].Type) {
+	case "aws":
+		ctrl, err = defaultoperation.NewDefaultSyncPut(opts,
+			armrpc_controller.ResourceOptions[datamodel.AWSPlane]{
+				RequestConverter:  converter.AWSPlaneDataModelFromVersioned,
+				ResponseConverter: converter.AWSPlaneDataModelToVersioned,
+			})
+
+	case "azure":
+		ctrl, err = defaultoperation.NewDefaultSyncPut(opts,
+			armrpc_controller.ResourceOptions[datamodel.AzurePlane]{
+				RequestConverter:  converter.AzurePlaneDataModelFromVersioned,
+				ResponseConverter: converter.AzurePlaneDataModelToVersioned,
+			})
+
+	case "radius":
+		ctrl, err = defaultoperation.NewDefaultSyncPut(opts,
+			armrpc_controller.ResourceOptions[datamodel.RadiusPlane]{
+				RequestConverter:  converter.RadiusPlaneDataModelFromVersioned,
+				ResponseConverter: converter.RadiusPlaneDataModelToVersioned,
+			})
+	}
```

Comment:
Do we need a default case here?

Reply:
I will add a default case that returns an error.